nilmtk/api.py

```diff
@@ -320,52 +369,69 @@ def call_predict(self,classifiers):
         
         pred_overall={}
         gt_overall={}
-        for name,clf in classifiers:
-            gt_overall,pred_overall[name]=self.predict(clf,self.test_mains,self.test_submeters, self.sample_period,'Europe/London')
+        if self.site_only == True:
+            for name,clf in classifiers:
+                pred_overall[name] = self.predict(clf,self.test_mains, self.sample_period,'Europe/London')
```

Comment:
Here again, you have a lot of duplicated code because you changed the signature of `self.predict()`.
Also, one method argurment is missing (`test_submeter`) so it won't work.

My suggestion: you revert your changes of `self.predict()` and you move the `self.site_only` test to choose or not to plot the `gt_overall`.